iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -360,14 +372,16 @@ private Message ProcessMessage(Message message, bool hasMessage)
                 {
                     lock (_syncRoot)
                     {
-                        _messageQueue.TryDequeue(out message);
-                        message.LockToken = message.LockToken;
-                        if (_qos == QualityOfService.AtLeastOnce)
+                        if (_messageQueue.TryDequeue(out message))
                         {
-                            _completionQueue.Enqueue(message.LockToken);
-                        }
+                            message.LockToken = message.LockToken;
+                            if (_qos == QualityOfService.AtLeastOnce)
+                            {
+                                _completionQueue.Enqueue(message.LockToken);
```

Comment:
What does enqueuing the lock token mean?

Reply:
this is used while marking a c2d message as complete. (MQTT supports only completing messages, while on AMQP we can reject and abandon as well)

Reply:
I mean, I generally think of enqueuing objects to be picked up by something on the other side. In this case, we're enqueuing the lock token, not an actually object (message).

Is this how we communicate back to the service that we complete a message, by enqueuing the lock token?